### PR TITLE
Validate documents on submission

### PR DESCRIPTION
**Pull request by nathanathan**
This adds schema validation for article/documents. It also fixes a bug where unresolved geoname ids would trigger exceptions instead of being ignored, and it might help with memory leak we saw on the server by using setInterval instead of recursive setTimeout calls.

### CODE REVIEW

File: server/methods.coffee [hunk 1]
```diff
@@ -67,14 +67,15 @@ Meteor.methods
           featureClass: geoname.featureClass
           featureCode: geoname.featureCode
           alternateNames: geoname.alternateNames
-      locationAnnotations = locationAnnotations.filter (loc)->
+      locationAnnotations.forEach (loc)->
         geoname = geonamesById[loc.geoname.geonameid]
         if geoname
           loc.geoname = geoname
```
Comment (jgoley) [inline server/methods.coffee]: Wondering if this is this necessary? Maybe we just do what's in the else block if there's no `geoname`?

Comment (nathanathan) [inline server/methods.coffee]: It is necessary. The geoname in the grits-api enhancement is different from the one that the geoname lookup service returns. We need the labels administrative division labels from it.

Comment (jgoley) [inline server/methods.coffee]: Is the lookup version supposed to replace the version in the original `enhancements` object? How is it being replaced?

Comment (nathanathan) [inline server/methods.coffee]: That's right. The original version is replaced by setting it to the look-up version stored in the geonamesById map.

**Comment by jgoley** [comment]
👍

**Pull request merged**